gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -131,30 +132,24 @@ def __repr__(self):
                                                  self.alts,
                                                  self.formats,
                                                  self.contigs]])
-
-  def _values_asdict(self, header):
-    """Converts PyVCF header values to ordered dictionaries."""
-    ordered_dict = collections.OrderedDict()
-    for key in header:
-      # These methods were not designed to be protected. They start with an
-      # underscore to avoid conflicts with field names. For more info, see
-      # https://docs.python.org/2/library/collections.html#collections.namedtuple
-      ordered_dict[key] = header[key]._asdict()  # pylint: disable=W0212
-    return ordered_dict
-
   def _get_infos_pysam(self, infos):
+    self.verify_infos(infos)
     results = collections.OrderedDict()
     for item in infos.items():
       result = collections.OrderedDict()
       result['id'] = item[0]
-      result['num'] = item[1].number if item[1].number != '.' else None
-      result['type'] = item[1].type
+      result['num'] = item[1].number
+      result['type'] = item[1].record['Type']
```

Comment:
Let's add enough checks making sure this field exist, similar to what you do below for 'source' and 'version'. Also please make the same change to `_get_formats_pysam()` function.